### PR TITLE
Swapper router tests additions

### PR DESCRIPTION
**Pull request by oldchili**

### CODE REVIEW

File: test/funnels/automation/StableSwapper.t.sol [hunk 1]
```diff
@@ -7,9 +7,13 @@ import { StableSwapper } from "src/funnels/automation/StableSwapper.sol";
...
 import { AllocatorBuffer } from "src/AllocatorBuffer.sol";
-import { TestUtils } from "test/utils/TestUtils.sol";
 
-contract StableSwapperTest is DssTest, TestUtils {
+contract StableSwapperTest is DssTest {
+    event Kissed (address indexed usr);
+    event Dissed (address indexed usr);
+    event Permit (address indexed usr);
+    event Forbid (address indexed usr);
+    event Config (address indexed src, address indexed dst,StableSwapper.PairConfig data);
```
Comment (sunbreak1211) [inline test/funnels/automation/StableSwapper.t.sol]:
```suggestion
    event Config (address indexed src, address indexed dst, StableSwapper.PairConfig data);
```

File: test/funnels/automation/StableSwapper.t.sol [hunk 2]
```diff
@@ -82,38 +84,139 @@ contract StableSwapperTest is DssTest, TestUtils {
...
+        emit Forbid(testAddress);
+        stableSwapper.forbid(testAddress);
+        assertEq(stableSwapper.bots(testAddress), 0);
+
+        stableSwapper.diss(address(this));
+
+        vm.expectRevert("StableSwapper/non-facilitator");
+        stableSwapper.permit(testAddress);
+        vm.expectRevert("StableSwapper/non-facilitator");
+        stableSwapper.forbid(testAddress);
```
Comment (sunbreak1211) [inline test/funnels/automation/StableSwapper.t.sol:127]: The new standard would say this should be checked in testModifiers.

File: test/funnels/automation/StableSwapper.t.sol [hunk 3]
```diff
@@ -82,38 +84,139 @@ contract StableSwapperTest is DssTest, TestUtils {
...
+    }
+
+    function testSetConfigByNonFacilitator() public {
+        assertEq(stableSwapper.buds(address(this)), 0);
+        vm.expectRevert("StableSwapper/non-facilitator");
+        emit Config(address(0x123), address(0x456), StableSwapper.PairConfig({
+            count: 23,
+            lot: uint96(314),
+            reqOut: uint112(42)
+        }));
```
Comment (sunbreak1211) [inline test/funnels/automation/StableSwapper.t.sol:157]: Remove this.

Comment (sunbreak1211) [inline test/funnels/automation/StableSwapper.t.sol]:
```suggestion
```

File: test/funnels/Depositor.t.sol [hunk 4]
```diff
@@ -156,6 +155,26 @@ contract DepositorTest is DssTest, TestUtils {
         assertEq(NftLike(UNIV3_POS_MGR).balanceOf(address(buffer)), 1);
     }
 
+    function testDepositAfterHop() public {
```
Comment (sunbreak1211) [inline test/funnels/Depositor.t.sol]: I think we should probably discard the changes in this file so we can only focus on complete the Swapper part as the Depositor will probably change anyway.

Comment (oldchili) [reply test/funnels/Depositor.t.sol]: Done.
The changes are now only part of https://github.com/makerdao/dss-allocator/pull/15.

File: test/funnels/automation/StableSwapper.t.sol (hunk 2 above)
Comment (oldchili) [reply test/funnels/automation/StableSwapper.t.sol:127]: Are you sure? can you give an example?

Comment (sunbreak1211) [reply test/funnels/automation/StableSwapper.t.sol:127]: Well we have been using `testModifiers`, not sure if calling it a new standard is accurate though. To be honest not strong feeling here.

**Review by sunbreak1211 — APPROVED** [review]
This should be good to merge, we can discuss the modifiers thing in the main PR.

**Review by telome — APPROVED** [review]
Good to merge

**Pull request merged**